libraries/AP_GPS/AP_GPS_UBLOX.cpp

```diff
@@ -498,6 +498,10 @@ AP_GPS_UBLOX::_request_next_config(void)
             const config_list *list = config_M10;
             const uint8_t list_length = ARRAY_SIZE(config_M10);
             Debug("Sending M10 settings");
+
+            // don't mix F9 and M10
+            _unconfigured_messages &= ~CONFIG_F9;
```

Comment:
I can't help but wonder whether a tweak to the code affected by the patch which introduced the problems might be better.  In particular:


```
@@ -334,6 +336,12 @@ AP_GPS_UBLOX::_request_next_config(void)
         }
         break;
     case STEP_POLL_GNSS:
+        if (supports_F9_config()) {
+            if (last_configured_gnss != params.gnss_mode) {
+                _unconfigured_messages |= CONFIG_F9;
+            }
+            break;
+        }
         if (!_send_message(CLASS_CFG, MSG_CFG_GNSS, nullptr, 0)) {
             _next_message--;
         }
```

Renaming "supports_f9_config" to "supports_valget_and_valset()` might stop this happening again?